pkg/registry/endpoint/etcd/etcd.go

```diff
@@ -33,7 +33,7 @@ type REST struct {
 }
 
 // NewStorage returns a RESTStorage object that will work against endpoints.
-func NewStorage(h tools.EtcdHelper) *REST {
+func NewStorage(s tools.StorageInterface) *REST {
```

Comment:
BTW - @thockin do you have any proposal what names those should have (I'm very bad in "inventing" names)